packages/legend-graph/src/models/protocols/pure/v1/engine/V1_EngineServerClient.ts

```diff
@@ -286,6 +290,21 @@ export class V1_EngineServerClient extends AbstractServerClient {
       { skipProcessing: Boolean(returnResultAsText) },
     );
 
+  doMappingTest = (
+    input: PlainObject<V1_ExecuteMappingTestInput>,
+    returnResultAsText?: boolean,
+  ): Promise<PlainObject<V1_MappingTestResult> | Response> =>
```

Comment:
Granted have not reviewed legend engine changes yet (will do asap)
But the API should return a lsit of `MappingTestResult` since it should be able to run, all, some or one tests. 
Therefore the return type should be `Promise<PlainObject<V1_MappingTestResult>[]>`